File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
       vector<int> ans;
       int n = nums.size();
       unordered_map<int,int> freq;
       priority_queue<pair<int,int>, vector<pair<int,int>>, greater<pair<int,int>>> pq;
       for(auto i : nums){
           freq[i]++;
       }

       for(auto i : freq){
           if(pq.size() < k ) {
             pq.push({i.second,i.first});
           }else {
             if(pq.top().first < i.second){
                pq.pop();
                pq.push({i.second,i.first});
             }
           }
        //    cout<<pq.size()<<endl;
       }

       while(!pq.empty()){
        ans.push_back(pq.top().second);
        pq.pop();
       }
       return ans; 
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] lSum = new int[nums.length];
        int[] rSum = new int[nums.length];
        int[] ans = new int[nums.length];
        for(int i = 0 ; i < nums.length ; i++){
            lSum[i] = i ==  0 ? 1 : nums[i-1]*lSum[i-1]; 
        }
        for(int i = nums.length -1  ; i >= 0  ; i--){
            rSum[i] = i ==  nums.length-1 ? 1: nums[i+1]*rSum[i+1]; 
        }

        for(int i = 0 ; i <  nums.length ; i++){
                ans[i] = lSum[i]*rSum[i];
        }
        return ans;
    }
}  
